packages/truffle-debugger/lib/solidity/reducers.js

```diff
@@ -77,14 +77,24 @@ const info = combineReducers({
   sourceMaps
 });
 
-export function functionDepth(state = 0, action) {
+export function functionDepthStack(state = [0], action) {
   switch (action.type) {
     case actions.JUMP:
+      let newState = state.slice(); //clone the state
       const delta = spelunk(action.jumpDirection);
-      return state + delta;
+      let top = newState[newState.length - 1];
+      newState[newState.length - 1] = top + delta;
```

Comment:
While we're at it, can we make the state look like:
```javascript
functionDepthStack: [{ depth: n }]
```

Or even rename `functionDepthStack` to something like `functionStack` or `jumpStack`? Then we can leave `functionDepth` as a name for the corresponding property of the stack frame, and that property can just remain an integer.

Reply:
I dunno, that just seems overcomplicated to me.  I'm having a hard time thinking of what else might potentially go on this particular stack.  Yes it's likely we'll eventually want a more proper Solidity stack, that we don't currently have, but this isn't the start of that -- remember, the elements of this stack represent EVM stackframes, not Solidity stackframes.